### PR TITLE
opensnitch, opensnitch-ui: 1.3.6 > 1.4.3

### PR DESCRIPTION
**Pull request by onny**
###### Motivation for this change
In this PR I would like to address following bug https://github.com/NixOS/nixpkgs/issues/148168 and continue the work of this PR https://github.com/NixOS/nixpkgs/pull/118329

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
Somehow running `opensnitch-ui` fails because it can't find `ui_pb2.py` in opensnitch python package.
```
$ opensnitch-ui
Traceback (most recent call last):
  File "/nix/store/qbqbsb1zbzbcrf421pdrby3qa7qj4rq5-opensnitch-ui-1.4.2/bin/.opensnitch-ui-wrapped", line 23, in <module>
    from opensnitch.service import UIService
  File "/nix/store/qbqbsb1zbzbcrf421pdrby3qa7qj4rq5-opensnitch-ui-1.4.2/lib/python3.9/site-packages/opensnitch/service.py", line 13, in <module>
    from opensnitch import ui_pb2
ImportError: cannot import name 'ui_pb2' from 'opensnitch' (/nix/store/qbqbsb1zbzbcrf421pdrby3qa7qj4rq5-opensnitch-ui-1.4.2/lib/python3.9/site-packages/opensnitch/__init__.py)
```
The file `ui_pb2` is missing but don't know why:
```
ls /nix/store/qbqbsb1zbzbcrf421pdrby3qa7qj4rq5-opensnitch-ui-1.4.2/lib/python3.9/site-packages/opensnitch
config.py  customwidgets.py  database.py  desktop_parser.py  dialogs  __init__.py  nodes.py  __pycache__  res  service.py  utils.py  version.py
```

**Comment by raboof** [comment]
I think that's a file that should be generated, probably at build time. Perhaps it used to be checked into version control and is no longer, or perhaps the build procedures changed and you need to explicitly run the generator.

**Comment by raboof** [comment]
That might be what https://github.com/NixOS/nixpkgs/pull/118329/files#diff-dda7ca3a66bb296c05bcc933df2aae7a511f317b3252d3d5c9b0c986edea36b1R32 was for?

**Comment by onny** [comment]
Good catch that worked!

**Comment by onny** [comment]
Hm next issue I'll have to look into, building opensnitch daemon
```
[...]
Building subPackage .
go: inconsistent vendoring in /build/source/daemon:
        golang.org/x/net@v0.0.0-20190311183353-d8887717615a: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/sys@v0.0.0-20190606203320-7fc4e5ec1444: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/net@v0.0.0-20191028085509-fe3aa8a45271: is marked as explicit in vendor/modules.txt, but not explicitly required in go.mod
        golang.org/x/sys@v0.0.0-20191029155521-f43be2a4598c: is marked as explicit in vendor/modules.txt, but not explicitly required in go.mod

        To ignore the vendor directory, use -mod=readonly or -mod=mod.
        To sync the vendor directory, run:
                go mod vendor
```

**Comment by onny** [comment]
Fixed it, builds fine now. If everyone wonders on how to fix "inconsistent vendoring" with building go modules, here's what I did:
```
git clone https://github.com/evilsocket/opensnitch.git
cd opensnitch/daemon
nix-shell -p protobuf go-protobuf protoc-gen-go-grpc
make -C ../proto ../daemon/ui/protocol/ui.pb.go
go mod tidy
go mod vendor
go mod install -mod=vendor
git diff --patch go.mod > go-mod.patch
```
I used this patch file in the package script and it works now :)

File: pkgs/tools/networking/opensnitch/ui.nix [hunk 1]
```diff
@@ -6,16 +6,19 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "opensnitch-ui";
-  version = "1.3.6";
+  version = "1.4.2";
```
Comment (raboof) [inline pkgs/tools/networking/opensnitch/ui.nix:9]: shouldn't this also be upped to 1.4.3?

Comment (onny) [reply pkgs/tools/networking/opensnitch/ui.nix]: Fixed :)

File: pkgs/tools/networking/opensnitch/go-mod.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+-	github.com/vishvananda/netns v0.0.0-20191106174202-0a2b9b5464df // indirect
+-	golang.org/x/net v0.0.0-20190311183353-d8887717615a
+-	golang.org/x/sync v0.0.0-20200625203802-6e8e738ad208 // indirect
+-	golang.org/x/sys v0.0.0-20190606203320-7fc4e5ec1444 // indirect
+-	golang.org/x/text v0.3.0 // indirect
++	golang.org/x/net v0.0.0-20191028085509-fe3aa8a45271
++	golang.org/x/sys v0.0.0-20191029155521-f43be2a4598c
+ 	google.golang.org/grpc v1.27.0
+ 	google.golang.org/protobuf v1.26.0
+ )
```
Comment (raboof) [inline pkgs/tools/networking/opensnitch/go-mod.patch:24]: Should we (also) propose such a change upstream, or is it no longer a problem on the main branch?

**Review by raboof — APPROVED** [review]

File: pkgs/tools/networking/opensnitch/go-mod.patch (hunk 2 above)
Comment (onny) [reply pkgs/tools/networking/opensnitch/go-mod.patch:24]: Haven't checked master or release candidate version. It is of course something wrong with the go.mod file upstream. Maybe with the next version it is fixed :)

**Pull request merged**